### PR TITLE
Update core to v0.89.7

### PR DESCRIPTION
**Pull request by beeender**

### CODE REVIEW

**Comment by beeender** [comment]
@realm/java YEAH

**Comment by kneth** [comment]
Please say "Realm's underlying storage engine" instead of "Realm Core". And "openssl" -> "OpenSSL".

Otherwise :+1:

**Comment by beeender** [comment]
@kneth We are always using "Realm core" in change log... Shall we start using another term from this time?

**Comment by kneth** [comment]
@beeender Right, keep it as "Realm core" (non-capital c)

File: changelog.txt [hunk 1]
```diff
@@ -3,6 +3,9 @@
  * Fixed a bug which might cause failure when loading the native library.
  * Fixed a bug which might trigger a timeout in Context.finalize().
  * Fixed a bug which might cause RealmObject.isValid() to throw an exception if the object is deleted.
+ * Updated Realm core to version 0.89.7
+   - fixed a potential stack overflow issue which might cause a crash when encryption was used.
```
Comment (cmelchior) [inline changelog.txt]: f -> F

File: changelog.txt [hunk 2]
```diff
@@ -3,6 +3,9 @@
...
+   - embedded crypto functions from OpenSSL into Realm dynamic lib to avoid random issues on some devices.
```
Comment (cmelchior) [inline changelog.txt]: e -> E

**Comment by cmelchior** [comment]
:+1:

File: changelog.txt (hunk 1 above)
Comment (nhachicha) [inline changelog.txt]: we can add _for some devices_ 
looking at you LG G3

**Comment by nhachicha** [comment]
:+1:

**Pull request merged**